File: ms-auth/src/main/java/infrastructure/configuration/BeanConfiguration.java
package infrastructure.configuration;

import application.port.input.UserRepositoryPort;
import application.port.output.TokenServicePort;
import application.service.CreateUserService;
import application.service.LoginService;
import application.usecase.CreateUserUseCase;
import application.usecase.LoginUsaCase;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.crypto.password.PasswordEncoder;

@Configuration
public class BeanConfiguration {
    @Bean
    public CreateUserUseCase createUserUseCaseBean(UserRepositoryPort userRepositoryPort) {
        return new CreateUserService(userRepositoryPort);
    }

    @Bean
    public LoginUsaCase loginUseCaseBean(UserRepositoryPort userRepositoryPort, TokenServicePort tokenServicePort, PasswordEncoder passwordEncoder) {
        return new LoginService(userRepositoryPort, tokenServicePort, passwordEncoder);
    }
}

File: ms-suggestions/src/main/kotlin/infrastructure/configuration/BeanConfiguration.kt
package infrastructure.configuration

import application.service.AiRecommenderService
import application.port.output.AiRecommenderPort
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

/**
 * Spring bean configuration for the application wiring.
 *
 * Provides beans that compose the use case implementations with their
 * required outbound ports.
 */
@Configuration
class BeanConfiguration {

    /**
     * Create the AiRecommenderService bean wired with the provided port.
     *
     * @param aiRecommenderPort outbound port implementation used by the service
     * @return a configured AiRecommenderService instance
     */
    @Bean
    fun aiRecommenderUseCase(aiRecommenderPort: AiRecommenderPort): AiRecommenderService {
        return AiRecommenderService(aiRecommenderPort = aiRecommenderPort)
    }
}